### DISCUSSION TITLE
Documentation and examples

### DISCUSSION THREAD

**Discussion by peacechen** [material-table-core/core#16] [General]
Mehmet published a terrific set of examples on his documentation site:
https://material-table.com/#/

Eventually there will be new features and/or changes to existing props and we should update the documentation. Does anyone have the source to the site above? It would be a shame to lose all of that and have to recreate it.

I suggest we petition material-ui to include this library. It would make this project and documentation more sustainable long term. This project adds the features needed for real world production use cases and would be a compelling addition to material-ui.
@mui-org @oliviertassinari @dmtrKovalenko ☝️

**Comment by oliviertassinari** [comment]
@peacechen You can find https://material-ui.com/components/data-grid/, it's meant to solve the same problem.

**Reply by peacechen** [reply]
I've tried the data-grid component but it's nowhere near as feature complete as material-table. Take a look at the documentation and examples page linked above. There's really no comparison.

**Reply by oliviertassinari** [reply]
What's the main feature missing?

**Reply by peacechen** [reply]
At the time I evaluated DataGrid, it didn't appear to support remote paginated data. Large data fetches (e.g. thousands of rows) is too slow across the network and remote data with pagination is a critical feature for our app. I see a paginationMode prop for DataGrid now, though it's not clear how the server calls are set up in that mode.

Integrated table search is great. Detail panel, editing, custom cell/row/column rendering, etc. While it's true users could build those features themselves, I expect a production grade table to offer a comprehensive set of boilerplate features.
There's an extensive set of examples that are very helpful to get started using it.
https://material-table.com/

XGrid looks like an interesting option. I gather that's where you prefer to steer users? The material-ui project could benefit from a quality table library, but if the table section is too crowded, we'll happily continue along separately.

**Reply by oliviertassinari** [reply]
Thanks for the feedback. Yeah, agree, date fetching is important. You can find a demo in https://material-ui.com/components/data-grid/pagination/#server-side-pagination.

The features you are referring to are planned to be available in the MIT version (search, detail panel, editing, custom rendering), you can [see the differentiators](https://deploy-preview-779--material-ui-x.netlify.app/components/data-grid/getting-started/#feature-comparison).

**Reply by peacechen** [reply]
I look forward to seeing those new features implemented. It could be beneficial to combine resources, but it's good to have choices as well.

**Reply by oliviertassinari** [reply]
@peacechen Yeah, having multiple options is great. It will take [a couple of months](https://npm-stat.com/charts.html?package=@material-ui/data-grid,material-table) for the new `DataGrid` to gain more usage by the community. At the end of the day, we don't really mind as we know that kinds of components can't be sustained only by an open-source model, very far from it, at least for where we want to push it. I estimate that it takes between $1m and $5m of R&D and a couple of years to build a world-class data grid component (I most likely underestimate the cost).

**Reply by peacechen** [reply]
I agree that a production grade table takes significant effort. I also believe that a well organized open source project could compete with the best of the commercial products. The reason being that open source allows anyone to contribute new features. Commercial projects are slaves to the almighty dollar and features deemed niche may never get implemented. Of course open source has its share of non-trivial problems. Recently I came across a service called Rysolv which allows people to pay bounties for features & bugfixes. That type of model could combine the best of both open source and commercial.

**Reply by matthewoestreich** [reply]
That is awesome @oliviertassinari! I haven't looked through the material-ui docs in about a year or so. I don't believe there was `XGrid` back then which is how I landed on `material-table` - in fact, `material-table` [was referenced in material-ui docs](https://web.archive.org/web/20190403134212if_/https://material-ui.com/demos/tables/#complementary-projects) at one point (and `DataGrid` looks like it has come a long way since then, too).

Awesome stuff!

**Reply by matthewoestreich** [reply]
@oliviertassinari should we continue this project or do you recommend that we contribute to material-ui?

**Reply by oliviertassinari** [reply]
@oze4 I would recommend either contributing to `material-table` or the `DataGrid` component.

**Reply by peacechen** [reply]
The more I use this table, the more impressed I am with it. I needed a column selector in my project and thought I would need to create it myself. To my delight, material-table has this built in through the `columnsButton` prop. Additionally, there was a recent bug fix related to column width definitions in this project that I needed as well. This has great community support.

@oliviertassinari This project is much further ahead than DataGrid. It will probably take a year for that to catch up, and I suspect it may never have all the features that this project already supports. `material-table` is a much better candidate for the material-ui project. It's silly to reinvent this wheel for the umpteenth time. What do you think about promoting XGrid as the commercial solution, and material-table as the FOSS option? DataGrid could remain as a lightweight option or an academic example.

**Comment by matthewoestreich** [comment]
A while back I attempted to simplify demos https://material-table-core.github.io/examples but over the weekend I have been moving all examples to https://material-table-core.com/demos

I think having a wide variety of demos for people to look through is huge, especially with such a flexible data grid component.

Thoughts?

**Reply by peacechen** [reply]
That's a great start @oze4 👍 
How do people contribute to it?

**Reply by matthewoestreich** [reply]
I need to write up docs on how to add demos..

- `https://material-table-core.com/demos` can be found over at https://github.com/material-table-core/website 
  - Uses [Docusaurus 2](https://v2.docusaurus.io/)
- `https://material-table-core.github.io/examples` can be found at https://github.com/material-table-core/examples
  - Home grown React app
  - Hoping to do away with this repo
  - Only after moving all demos to the `material-table-core/website` repo

**Reply by matthewoestreich** [reply]
@peacechen I have started documenting this [here](https://material-table-core.com/demos#how-to-add-demo)
